File: android/app/src/main/java/com/smartexpense/android/domain/usecase/user/GetMeUseCase.java
package com.smartexpense.android.domain.usecase.user;

import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto;
import com.smartexpense.android.domain.repository.ResultCallback;
import com.smartexpense.android.domain.repository.UserRepository;

public class GetMeUseCase {
    private final UserRepository repository;

    public GetMeUseCase(UserRepository repository) {
        this.repository = repository;
    }

    public void execute(ResultCallback<UserProfileResponseDto> callback) {
        repository.getMe(callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/user/GetMeUseCase.kt
package com.smartexpense.android.domain.usecase.user

import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto
import com.smartexpense.android.domain.repository.UserRepository

class GetMeUseCase(private val repository: UserRepository) {
    suspend fun execute(): Result<UserProfileResponseDto> {
        return repository.getMe()
    }
}
